lib/components/kernel-monitor.js

```diff
@@ -66,22 +54,108 @@ const KernelMonitor = observer(({ store }: { store: store }) => {
       </ul>
     );
   }
-  const grouped = _.groupBy(
-    store.runningKernels,
-    kernel => kernel.gatewayName || "Local"
-  );
-  return (
-    <div className="kernel-monitor">
-      {_.map(grouped, (kernels, group) => (
-        <MonitorSection
-          store={store}
-          kernels={kernels}
-          group={group}
-          key={group}
-        />
-      ))}
-    </div>
-  );
+
+  const data = _.map(store.runningKernels, kernel => {
+    return {
+      gateway: kernel.gatewayName || "Local",
+      displayName: kernel.displayName,
+      status: kernel.executionState,
+      executionCount: kernel.executionCount,
+      lastExecutionTime: kernel.lastExecutionTime,
+      kernel: kernel,
+      files: store.getFilesForKernel(kernel)
+    };
+  });
+  const columns = [
+    {
+      Header: "Gateway",
+      accessor: "gateway",
+      maxWidth: 150
+    },
+    {
+      Header: "Kernel",
+      accessor: "displayName",
+      maxWidth: 150
+    },
+    {
+      Header: "Status",
+      accessor: "status",
+      maxWidth: 50
+    },
+    {
+      Header: "Count",
+      accessor: "executionCount",
+      maxWidth: 50,
+      style: { textAlign: "right" }
+    },
+    {
+      Header: "Last Exec Time",
+      accessor: "lastExecutionTime",
+      maxWidth: 100,
+      style: { textAlign: "right " }
+    },
+    {
+      Header: "Managements",
+      accessor: "kernel",
+      Cell: props => {
+        const kernel = props.value;
+        return [
+          <a
+            className="icon icon-playback-pause"
+            onClick={interrupt.bind(this, kernel)}
+            title="Interrupt kernel"
+            key={kernel.displayName + "interrupt"}
+          />,
+          <a
+            className="icon icon-sync"
+            onClick={restart.bind(this, kernel)}
+            title="Restart kernel"
+            key={kernel.displayName + "restart"}
+          />,
+          <a
+            className="icon icon-trashcan"
+            onClick={shutdown.bind(this, kernel)}
+            title="Shutdown kernel"
+            key={kernel.displayName + "shutdown"}
```

Comment:
I didn't encounter any problem or error message maybe because within react-table, the row itself may have an unique key ?

Anyway, I changed it to use index as a key like: `map(runningKernels, (kernel, key: number) => {...}`, which must be unique across the row & cells